third_party/nvfuser/python_tests/test_python_frontend.py

```diff
@@ -662,10 +662,42 @@ def fusion_func(fd: FusionDefinition) :
             with FusionDefinition() as fd:
                 fusion_func(fd)
             raise RuntimeError("FusionDefinition.from_pytorch should have raised an error for a CPU Tensor!")
-        except TypeError:
+        except ValueError:
             pass
-        except RuntimeError:
-            raise
+
+    def test_no_definition(self):
+        inputs = [
+            torch.randn(4, 4, device='cpu'),
+        ]
+
+        # A FusionDefinition object is constructed but not defined, should trip an error
+        try:
+            fd = FusionDefinition()
+            out = fd.execute(inputs)
+            raise RuntimeError("Expecting an error for a lack of a child class defining a definition!")
+        except NotImplementedError:
+            pass
+
+    def test_func_definition(self) :
+        inputs = [
+            torch.randn(4, 4, device='cuda'),
+        ]
+
+        def fusion_func(fd: FusionDefinition) :
+            t0 = fd.from_pytorch(inputs[0])
+            t1 = fd.ops.sigmoid(t0)
+            fd.add_output(t1)
+
+        class MyFusion(FusionDefinition):
+            def definition(self) :
+                t0 = fd.from_pytorch(inputs[0])
```

Comment:
should the `fd` here be `self` instead? I think we are using the fd that was defined below `fd = MyFusion()`. python can be confusing sometimes....

Reply:
Yes, I should have used self.  I am not clear why `fd` is working in this test.  If I play with the name and use `fs`, for instance, then it errors out as an undefined name.  It seems like even though `fd` is defined after the class is defined, it still acts as a closure with `fd` included.

Thanks for catching that I wrote a really sloppy test.